ruma-common/src/push/condition.rs

```diff
@@ -42,13 +52,273 @@ pub enum PushCondition {
     },
 }
 
+impl PushCondition {
+    /// Check if this condition applies to the event.
+    ///
+    /// # Arguments
+    ///
+    /// * `event` - The flattened JSON representation of a room message event.
+    /// * `context` - The context of the room at the time of the event.
+    pub fn applies(&self, event: &FlattenedJson, context: &PushConditionRoomCtx) -> bool {
+        match self {
+            Self::EventMatch { key, pattern } => {
+                let value = match key.as_str() {
+                    "room_id" => &context.room_id,
+                    _ => match event.get(key) {
+                        Some(v) => v,
+                        None => return false,
+                    },
+                };
+
+                value.matches_pattern(pattern, key == "content.body")
+            }
+            Self::ContainsDisplayName => {
+                let value = match event.get("content.body") {
+                    Some(v) => v,
+                    None => return false,
```

Comment:
Actually, I've reviewed all cases again and `sender` not being a valid user ID is the only one that seems like it could truly be regarded as an error case. I won't open an issue about it, current API seems good.